src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,236 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.runtime.view.LayoutView;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+
+/**
+ * Instantiates and performs failure detection adn handling asynchronously.
+ * <p>
+ * Created by zlokhandwala on 9/28/16.
+ */
+@Slf4j
+public class ManagementServer extends AbstractServer {
+
+    @Override
+    public void reset() {
+    }
+
+    @Override
+    public void reboot() {
+    }
+
+    /**
+     * The options map.
+     */
+    private Map<String, Object> opts;
+    private ServerContext serverContext;
+
+    private IFailureDetectorPolicy failureDetectorPolicy;
+    private CorfuRuntime corfuRuntime;
+
+    /**
+     * Flag to notify management server of shutdown
+     */
+    @Getter
+    private boolean isShutdown = false;
+    /**
+     * Stores this current node's address
+     */
+    @Getter
+    private String localEndpoint;
+    /**
+     * Used for initial bootstrap only.
+     */
+    private LayoutServer layoutServer;
+    /**
+     * Interval in executing the failure detection policy.
+     * In milliseconds.
+     */
+    @Getter
+    private long policyExecuteInterval = 1000;
+    /**
+     * To schedule failure detection.
+     */
+    private final ExecutorService schedulerService;
+
+
+    public ManagementServer(ServerContext serverContext, LayoutServer layoutServer) {
```

Comment:
So as I have mentioned this in the above description, the layout server is required for initial bootstrapping. The management server needs to know which layout servers should it connect the runtime to. Once the runtime is connected, its no longer required.
Let me know if there is a workaround for this.

(I wanted to avoid giving the user another cmdlet to bootstrap the management server.)